conda_build/environ.py

```diff
@@ -47,6 +47,10 @@ def get_git_build_info(src_dir, git_url, expected_rev):
     env = os.environ.copy()
     d = {}
     git_dir = join(src_dir, '.git')
+    if not isinstance(git_dir, str):
+        # On Windows, subprocess env can't handle unicode.
+        git_dir = git_dir.encode(sys.getfilesystemencoding())
```

Comment:
Note that it is possible for the encoding to be `None` on Unix. ( https://docs.python.org/2/library/sys.html#sys.getfilesystemencoding ) This becomes an issue with docker containers for example.